src/coreclr/src/binder/bindertracing.cpp

```diff
@@ -80,16 +80,48 @@ namespace
 #endif // FEATURE_EVENT_TRACE
     }
 
-    void GetAssemblyLoadContextNameFromBindContext(ICLRPrivBinder *bindContext, AppDomain *domain, /*out*/ SString &alcName)
+    void GetAssemblyLoadContextNameFromManagedALC(INT_PTR managedALC, /* out */ SString &alcName)
     {
-        _ASSERTE(bindContext != nullptr);
+#ifdef CROSSGEN_COMPILE
+        alcName.Set(W("Custom"));
+#else // CROSSGEN_COMPILE
+        // Keeping a per-thread INT_PTR reference to a managed AssemblyLoadContext is
+        // sufficient to avoid repeatedly calling into managed code, as managedALC does not
+        // change between each resolve+bind operations.
+        static thread_local INT_PTR prevManagedALC;
```

Comment:
To be honest I'm not a fan of this solution. While it will probably work it feels too "fragile". There are basically two callers for this functionality:
* `BinderTracing PopulateBindRequest` which caches the results on its `BindRequest` object.
* The newly added `ResolutionAttemptedOperation` which can cache the results since it's also instance based.

Caching the results feel like a more natural solution to the problem. Not counting that with the change as is, one caller is caching and the other is not.